### DISCUSSION TITLE
Demo of a distributed search browser extension: IndieSearch!

### DISCUSSION THREAD

**Discussion by jphastings** [Pagefind/pagefind#685] [Show and tell]
A few months back I built a demo of a fully decentralised and self-owned search engine that runs from your browser — built with the power of Pagefind! Check out [IndieSearch](https://github.com/jphastings/indiesearch)

https://github.com/jphastings/indiesearch/assets/42999/b1042abe-6aef-4157-bc08-770cfd3becfc

This demo shows:

- 🖱️ Accessing the search page direct from your extensions list
- 😔 Performing a search with no sites in your IndieSearch index (no results!)
- 🤓 (Cheatcode) Using a quick hack to add the PageFind site's search to the local index (these wouldn't be present in a proper build)
- 🔍 Searching for the word "hello" (Lots of hits of pagefind.app!)
- ❌ Marking that newly added site to be excluded from future searches
- 🤓 (Cheatcode) Adding a search index for the XKCD site to the local index
- 🔍 Searching for the word "hello" (includes XKCD but _not_ PageFind, as it's excluded)
- 🌐 Visiting my blog
- ✨ My site is automatically added to your search index as a new site (it has the magic header!)
- 🔍 Searching for the word "hello" (includes results from my blog and XKCD, but still not PageFind)

It's still very much a prototype (I haven't even shipped it to the browser extension stores) so I'd love to hear from people who are interested in this, to help me prioritise putting more work into it.

I also wrote a little more [on my blog](https://www.byjp.me/posts/indiesearch/) — but hadn't realised there was a show and tell section here to add to! ☺️

**Comment by bglw** [comment]
Hi @jphastings ! 👋 

Very nice! Keen to keen an eye on it 😄
